cosmic-panel-bin/src/space/wrapper_space.rs

```diff
@@ -340,295 +323,292 @@ impl WrapperSpace for PanelSpace {
-                                                applet_env.push((
-                                                    "X_PRIVILEGED_WAYLAND_SOCKET".to_string(),
-                                                    privileged_socket.as_raw_fd().to_string(),
-                                                ));
-                                                fds.push(privileged_socket.into());
-                                                *security_ctx = Some(security_context);
-                                            }
-                                            Err(why) => {
-                                                error!(?why, "Failed to create a listener");
-                                            }
-                                        }
-                                    };
-                                }
+            for (panel_client, my_list) in panel_clients {
+                if !panel_client.exec.is_some() {
+                    continue;
+                }
+                let Some(socket) = panel_client.stream.take() else {
+                    error!("Failed to get socket for {}", &panel_client.name);
+                    continue;
+                };
+
+                // Ensure there is only one applet per panel with minimize
+                panel_client.minimize_priority = if panel_client
+                    .minimize_priority
+                    .is_some_and(|x| x == max_minimize_priority && !has_minimize)
+                {
+                    has_minimize = true;
+                    Some(max_minimize_priority)
+                } else {
+                    None
+                };
 
-                                for (key, val) in &env_vars {
-                                    if !requests_wayland_display && *key == "WAYLAND_DISPLAY" {
-                                        continue;
-                                    }
-                                    applet_env.push((key.clone(), val.clone()));
-                                }
+                let is_notification_applet = panel_client.is_notification_applet.unwrap_or(false);
+                let requests_wayland_display =
+                    panel_client.requests_wayland_display.unwrap_or(false);
+
+                let mut exec_iter = Shlex::new(&panel_client.exec.as_deref().unwrap());
+                let exec = exec_iter
+                    .next()
+                    .expect("exec parameter must contain at least on word");
+
+                let mut args = Vec::new();
+                for arg in exec_iter {
+                    trace!("child argument: {}", &arg);
+                    args.push(arg);
+                }
+                let mut fds = Vec::with_capacity(2);
+                let mut applet_env = Vec::new();
+
+                applet_env.push((
+                    "X_MINIMIZE_APPLET".to_string(),
+                    if panel_client.minimize_priority.is_some() {
+                        "true".to_string()
```

Comment:
`true.to_string()` and `false.to_string()` should be ok here.

Reply:
or just `panel_client.minimize_priority.is_some().to_string()`

Reply:
man do I love rust. Fixed